src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/Program.cs

```diff
@@ -10,10 +13,119 @@ namespace SharedPoolsOfWCFObjects
 {
     public class Program
     {
-        public void Main(string[] args)
+        private enum Program2Run { Stress, Perf };
+
+        // Parameters and default values
+        private Program2Run _paramProgram2Run = Program2Run.Stress;
+        private int _paramStressLevel = DefaultStressLevel;
+        private TimeSpan _paramStressRunDuration = TimeSpan.FromMinutes(1);
+        private string _paramHostName = "localhost";
+        private TestBinding _paramBinding = TestBinding.Http;
+        private bool _paramUseAsync = true;
+        private string _paramTestToRun = "HelloWorld";
+
+        private const int DefaultStressLevel = 10;
+        private const long DefaultStressIterations = 1024L * 1024L * 1024L * 1024L;
+        // Sometimes it is beneficial to execute stress tests a certain number of iterations rather than run for a certain period of time.
+        // The default number of iterations is an arbitrary large number that would leave this type of stress running for a very long time.
+        private long _paramIterations = DefaultStressIterations;
+
+        public static void Main(string[] args)
+        {
+            var test = new Program();
+            test.ProcessRunOptions(args);
+            if (test._paramProgram2Run == Program2Run.Stress)
+            {
+                test.RunStress();
+            }
+            else
+            {
+                test.RunPerf();
+            }
+        }
+
+        private bool ProcessRunOptions(string[] args)
+        {
+            //var paramHostName = Environment.GetEnvironmentVariable("HostName");
```

Comment:
Nit: it would be nice to eliminate these commented out sections.  Or if you occasionally use them for private testing, put them under a conditional compile check.